server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -291,7 +291,6 @@ static int determineMaximumNumberOfSlices(
         ToLongFunction<String> fieldCardinality
     ) {
         return executor instanceof ThreadPoolExecutor tpe
-            && tpe.getQueue().isEmpty()
```

Comment:
@javanna I assume your recent benchmark experiments show that this should stay removed. Just double checking since this removal stuck out a bit and I saw it was just removed some days ago.

Reply:
yes, I need to work out some kind of replacement for it, but I will that that in main once this PR is merged.